File: packages/sea-core-ios/Tests/SEACoreTests/SEAScreenSecurityTests.swift
import XCTest
@testable import SEACore

final class SEAScreenSecurityTests: XCTestCase {
    func test_notCaptured_neverProducesAnOverlay_regardlessOfPolicy() {
        for policy: SEACapturePolicy in [.log, .warn, .blockInput] {
            XCTAssertEqual(SEAScreenSecurity.action(forCaptured: false, policy: policy), .none)
        }
    }

    func test_captured_logPolicy_producesNoOverlay() {
        XCTAssertEqual(SEAScreenSecurity.action(forCaptured: true, policy: .log), .none)
    }

    func test_captured_warnPolicy_producesNonBlockingOverlay() {
        XCTAssertEqual(SEAScreenSecurity.action(forCaptured: true, policy: .warn), .overlay(blocksInput: false))
    }

    func test_captured_blockInputPolicy_producesBlockingOverlay() {
        XCTAssertEqual(SEAScreenSecurity.action(forCaptured: true, policy: .blockInput), .overlay(blocksInput: true))
    }
}

File: packages/sea-core-android/src/test/java/com/bankerise/sea/core/SEAScreenSecurityTests.kt
package com.bankerise.sea.core

import org.junit.Assert.assertEquals
import org.junit.Test

class SEAScreenSecurityTests {

    @Test
    fun `not captured never produces overlay regardless of policy`() {
        for (policy in SEACapturePolicy.entries) {
            assertEquals(
                SEACaptureAction.None,
                SEAScreenSecurity.evaluateAction(isCaptured = false, policy = policy),
                "policy=$policy"
            )
        }
    }

    @Test
    fun `captured with LOG policy produces no overlay`() {
        assertEquals(
            SEACaptureAction.None,
            SEAScreenSecurity.evaluateAction(isCaptured = true, policy = SEACapturePolicy.LOG)
        )
    }

    @Test
    fun `captured with WARN policy produces non-blocking overlay`() {
        val action = SEAScreenSecurity.evaluateAction(
            isCaptured = true,
            policy = SEACapturePolicy.WARN
        )
        assertEquals(SEACaptureAction.Overlay(blocksInput = false), action)
    }

    @Test
    fun `captured with BLOCK_INPUT policy produces blocking overlay`() {
        val action = SEAScreenSecurity.evaluateAction(
            isCaptured = true,
            policy = SEACapturePolicy.BLOCK_INPUT
        )
        assertEquals(SEACaptureAction.Overlay(blocksInput = true), action)
    }
}
